packages/api/models/granules.js

```diff
@@ -283,72 +283,72 @@ class Granule extends Manager {
   /**
    * Create new granule records from incoming sns messages
    *
-   * @param {Object} payload - sns message containing the output of a Cumulus Step Function
+   * @param {Object} cumulusMessage - a Cumulus Message
    * @returns {Promise<Array>} granule records
    */
-  async createGranulesFromSns(payload) {
-    const granules = get(payload, 'payload.granules', get(payload, 'meta.input_granules'));
-
-    if (!granules) return Promise.resolve();
-
-    const executionName = get(payload, 'cumulus_meta.execution_name');
-    const arn = aws.getExecutionArn(
-      get(payload, 'cumulus_meta.state_machine'),
-      executionName
+  async createGranulesFromSns(cumulusMessage) {
+    const granules = get(cumulusMessage, 'payload.granules')
+      || get(cumulusMessage, 'meta.input_granules');
+
+    if (!granules) return null;
+
+    const executionName = get(cumulusMessage, 'cumulus_meta.execution_name');
+    if (!isString(executionName)) return null;
+
+    const stateMachine = get(cumulusMessage, 'cumulus_meta.state_machine');
+    if (!isString(stateMachine)) return null;
+
+    const executionArn = aws.getExecutionArn(stateMachine, executionName);
+    const executionUrl = aws.getExecutionUrl(executionArn);
+
+    const collection = get(cumulusMessage, 'meta.collection');
+
+    return Promise.all(
+      granules
+        .filter((g) => g.granuleId)
+        .map(async (granule) => {
+          const granuleFiles = await buildDatabaseFiles({
+            providerURL: buildURL({
+              protocol: cumulusMessage.meta.provider.protocol,
+              host: cumulusMessage.meta.provider.host,
+              port: cumulusMessage.meta.provider.port
+            }),
+            files: granule.files
+          });
+
+          const temporalInfo = await cmrjs.getGranuleTemporalInfo(granule);
+
+          const doc = {
+            granuleId: granule.granuleId,
+            pdrName: get(cumulusMessage, 'meta.pdr.name'),
+            collectionId: constructCollectionId(collection.name, collection.version),
+            status: get(cumulusMessage, 'meta.status'),
+            provider: get(cumulusMessage, 'meta.provider.id'),
+            execution: executionUrl,
+            cmrLink: granule.cmrLink,
+            files: granuleFiles,
+            error: parseException(cumulusMessage.exception),
+            createdAt: get(cumulusMessage, 'cumulus_meta.workflow_start_time'),
+            timestamp: Date.now(),
+            productVolume: getGranuleProductVolume(granuleFiles),
+            timeToPreprocess: get(granule, 'sync_granule_duration', 0) / 1000,
+            timeToArchive: get(granule, 'post_to_cmr_duration', 0) / 1000,
+            processingStartDateTime: granule.sync_granule_end_time
+              ? (new Date(granule.sync_granule_end_time)).toISOString()
+              : undefined,
+            processingEndDateTime: granule.post_to_cmr_start_time
```

Comment:
If the granule is not published to cmr (no cmr step), I think we still need processingStartDateTime/processingEndDateTime.

Reply:
If these properties `sync_granule_end_time, sync_granule_duration, post_to_cmr_duration, and post_to_cmr_start_time` are used for populating `timeToPreprocess, timeToArchive, processingStartDateTime, processingEndDateTime`, should they be removed from granule when they are not needed anymore?

Reply:
> should they be removed from granule when they are not needed anymore?

This code is not run as part of a workflow, it is triggered by an execution being written to SNS.  The return value is not used anywhere, so there isn't really a reason to modify the input granule.

Reply:
> If the granule is not published to cmr (no cmr step), I think the granule should still have processingStartDateTime/processingEndDateTime. These two fields are required by EMS Ing report.

I talked to @bilts about it, and he felt that was out of scope for this ticket.